api/v1beta2/condition_types.go

```diff
@@ -71,6 +71,10 @@ const (
 	// required fields, or the provided credentials do not match.
 	AuthenticationFailedReason string = "AuthenticationFailed"
 
+	// SourceVerifiedFailedReason signals that the Source's verification
+	// check failed.
+	SourceVerifiedFailedReason string = "SourceVerificationFailed"
```

Comment:
What about "OCISourceSignatureVerifyFailedReason"? Does that make sense? I've updated the code according to this, please let me know if you want me to change it, thanks.

Reply:
No, to be more specific, I meant that in the last implementation `SourceVerificationFailed` reason was used for multiple failures where one was actual verification failed and other was related to verification material not found due to which verification can't be done. Instead, the reason could be more specific. Reusing `VerificationError` from GitRepo reconciler for actual verification failure and then a new, more specific, reason for missing verification material, call it something appropriate based on the context, can be used. Wherever possible, it'd be better to have common reasons, not make them reconciler specific.
Since there's no constant for `VerificationError` you should create one.

Reply:
thanks to @Dentrax, he addressed this in #890